### PR TITLE
Split preprocessing into a feature_engineering script

### PR DESCRIPTION
**Pull request by cmmclaug**
* Completed the cleaning of the name feature by splitting into first/last and removing universe stuff. Still kept name for now, can drop in new feature_engineering.py script
* Created new feature_engineering.py script that does the train/test/split now. No feature engineering here yet, but it's set up for Ivy to work on.
* Hooked up new script in makefile flow, and added a new comment header for Milestone 3 release.

### CODE REVIEW

**Comment by cmmclaug** [comment]
Oh, and also fixed the pickle model dump bug in analysis.py I was referring to earlier.

File: src/analysis.py [hunk 1]
```diff
@@ -238,7 +238,7 @@ def train_models(train_df, models, param_grid=None, output_dir=""):
             random_search.fit(X_train, y_train)
             if output_dir:
                 pickle.dump(random_search.best_estimator_, 
-                open(output_dir + "\models\optimized_model.pkl", 'wb'))
+                open(output_dir + "/models/optimized_model.pkl", 'wb'))
```
Comment (zmerpez) [inline src/analysis.py:241]: Nice catch!

File: src/feature_engineer.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    assert os.path.isfile(input_filename), "Invalid input filename path provided"
+    if not os.path.exists(os.path.dirname(output_filename)):
+        os.makedirs(os.path.dirname(output_filename))
+    assert os.path.exists(os.path.dirname(output_filename)), "Invalid output path provided"
+
+    output_file_slug = output_filename.split(".")[0]
+    output_file_ext = output_filename.split(".")[-1]
+
+    clean_df = pd.read_csv(input_filename)
+
```
Comment (zmerpez) [inline src/feature_engineer.py:38]: We may add a comment for future engineering happens at this line.

File: src/feature_engineer.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+Usage: feature_engineer.py -i=<input> -o=<output> [-v]
+
+Options:
+-i <input>, --input <input>     Local processed clean data csv file
+-o <output>, --output <output>  Local output filename and path for processed data csv
+[-v]                            Report verbose output of dataset retrieval process
+"""
+import os
+import glob
```
Comment (zmerpez) [inline src/feature_engineer.py]: are we going to use glob here?

**Review by zmerpez — APPROVED** [review]
Thanks !

**Review by ssyayayy — APPROVED** [review]
Looking good.

**Review by aidanmattrick — APPROVED** [review]
Looks good to me!

**Pull request merged**